### PR TITLE
`Programming exercises`: Allow instructors to clone the solution and test repos using an access token based HTTP URL

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
In Gitlab/Jenkins setups Artemis can create an access token so that users can clone repositories via HTTP without passwords by embedding the token as part of the URL.

### Description
The URLs as shown to the instructors for the template, solution, and test repositories did not show that token even if it existed for those users. Therefore, instructors would still have to either set a password in Gitlab and use that when checking out via HTTP or they would have to clone via SSH.
This PR enables the access token based cloning for the three exercise repositories as well.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

I‘m not sure if the VCS-access-tokens feature is enabled on TS2 (config option `artemis.version-control.version-control-access-token: true`). if so, you can test there, otherwise locally on a Gitlab/Jenkins setup.

1. Log in to Artemis as the Instructor
2. Start a programming exercise as if you were a student with your instructor account to make sure an access token has been created for your account.
   - Refresh the Artemis page with F5 to force a refresh of the user information including the newly created token from the server.
   - If the URL shown for ‘Clone Repository‘ in an existing participation is already like `https://username:****@…`, then the token exists and this step can be skipped. 
4. Go the the details page of a programming exercise.
5. Scroll down to the point where the three repo links for the template, solution, and test repositories are listed. The URL shown when clicking on the ‘Clone Repository’ button should contain your username and the token (hidden as `***`) as part of the URL.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| `clone-repo-button.component.ts` | 82.6% | 91.7% | 

### Screenshots
Before:
Screenshot 2022-07-01 at 11-51-03 Programming Exercises

After:
Screenshot 2022-07-01 at 11-50-35 Programming Exercises

### CODE REVIEW

File: src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts [hunk 1]
```diff
@@ -105,18 +105,15 @@ export class CloneRepoButtonComponent implements OnInit {
...
-        }
-
         const token = insertPlaceholder ? '**********' : vcsAccessToken;
-
-        const protocolAndUsername = repositoryUrlParts[0];
-        const host = repositoryUrlParts[1];
-
-        return protocolAndUsername + ':' + token + '@' + host;
+        const urlUserInfoPart = `://${this.user.login}:${token}@`;
+        if (url.indexOf('@') === -1) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts:110]:
```suggestion
        if (!url.includes('@')) {
```

**Review by dfuchss — APPROVED** [review]
Code change is small and looks good

**Review by pantageepapa — APPROVED** [review]
Tested on ts2: HESTIA Test Course - works as described

**Review by manuelmanso — APPROVED** [review]
Test on TS2 and works as expected. Code changes look good 👍

**Pull request merged**